### PR TITLE
fix(anchor-listener, core): batch get logs

### PR DESCRIPTION
**Pull request by stephhuynh18**
Get logs in the anchor-listener get logs for a range now (set to 100 blocks as default)

### CODE REVIEW

**Review by oed — COMMENTED** [review]
Seems like a good cleanup. 

1. What range are we using for queries? Iirc `eth_getLogs` can get really slow if you use a too large range. We likely need to cap it at some particular value. Not sure if the order of manitude is more like 10 or 100 or 1000.

2. What's the reason the loader needs to call `provider.getBlock(...)`? Does it call that for every block we call getLogs for?

**Comment by stephhuynh18** [comment]
> Seems like a good cleanup.
> 
> 1. What range are we using for queries? Iirc `eth_getLogs` can get really slow if you use a too large range. We likely need to cap it at some particular value. Not sure if the order of manitude is more like 10 or 100 or 1000.
> 2. What's the reason the loader needs to call `provider.getBlock(...)`? Does it call that for every block we call getLogs for?

@oed 
1. We are using 100 and can adjust if that is still too low. We can also make it configurable at a later time
2. The loader has a function to add block information to each batch of events. This is only used during a continuous sink to detect for reorgs (so on every NEW block). Historical syncs do not use this.

**Comment by oed** [comment]
@stephhuynh18 Thanks, sounds good!

**Comment by oed** [comment]
(I'll let someone else do the in depth code review)

File: packages/anchor-listener/src/loader.ts [hunk 1]
```diff
@@ -223,8 +251,13 @@ export function createBlocksProofsLoader({
 }: BlocksProofsLoaderParams): Observable<BlockProofs> {
   const retry = retryConfig ?? { count: 3 }
   return range(fromBlock, toBlock - fromBlock + 1).pipe(
-    bufferCount(blockLoadBuffer ?? 5),
-    mapLoadBlocksProofs(provider, chainId, retry)
+    bufferCount(blockLoadBuffer ?? 100),
```
Comment (stbrody) [inline packages/anchor-listener/src/loader.ts:254]: Can we make a constant for the batch size?

**Review by stbrody — APPROVED** [review]
jumping in here on the review since Paul is out.  This code is complex and a bit hard to follow, so would probably still benefit from another set of eyes on it, but as best as I can tell it LGTM

File: packages/anchor-listener/src/loader.ts [hunk 2]
```diff
@@ -59,97 +71,104 @@ export async function loadBlock(
...
+      address,
+      fromBlock: blockRangeFilter.fromBlock,
+      toBlock: blockRangeFilter.toBlock,
+    })
   }).pipe(
     retry(retryConfig),
-    map((logs) => {
-      return logs.map((log) => createAnchorProof(chainId, log))
+    map(groupLogsByBlockNumber),
+    map((logsByBlockNumber) => {
```
[the remark this answers is not in the document]
Comment (stephhuynh18) [reply packages/anchor-listener/src/loader.ts]: Yup good point
